data_collector.py

```diff
@@ -0,0 +1,136 @@
+"""
+Script for collecting telemetry data alongside video capture from Intel RealSense.
+
+usage: data_collector.py [-h] [--no-telem] [--no-save] [--no-display]
+
+options:
+  -h, --help    show this help message and exit
+  --no-telem    do not gather telemetry data
+  --no-save     do not store captured video
+  --no-display  do not display video capture
+"""
+
+import pyrealsense2 as rs
+import numpy as np
+import cv2
+import argparse
+import os
+import json
+
+from pathlib import Path
+from datetime import datetime
+
+from drone_control.mavlink_telemetry import MavlinkTelemetry
+
+FPS = 30
+RESOLUTION = (1280, 720)
+
+TELEMETRY_CONNECTION_STRING = "udpin:0.0.0.0:1455"
+
+datetime_str = datetime.now().strftime("%m-%d-%Y %H-%M-%S")
+DATA_STORE_DIR = Path(f"data/test_data/{datetime_str}")
+
+
+def data_collector(no_save: bool = True, no_telem: bool = False, no_display=False):
+    """Collects telemetry data alongside video capture.
+    Press Ctrl+C to stop.
+
+    :param no_save: bool, defaults to True, determines if the video capture should be saved.
+    :param no_telem: bool, defaults to False, determines if the telemetry data should be gathered.
+    :param no_display: bool, defaults to False, determines if the video capture should be displayed.
+    """
+    pipeline = rs.pipeline()
+    config = rs.config()
+
+    config.enable_stream(rs.stream.depth, 640, 480, rs.format.z16, 30)
+    config.enable_stream(rs.stream.color, 1280, 720, rs.format.bgr8, 30)
+
+    pipeline.start(config)
+
+    if not no_telem:
+        telemetry = MavlinkTelemetry()
+        telemetry.connect(TELEMETRY_CONNECTION_STRING)
+        telemetry.start()
+
+    if not no_save:
+        os.makedirs(DATA_STORE_DIR, exist_ok=True)
+        color_out = cv2.VideoWriter(
+            str(DATA_STORE_DIR / "color_out.mp4"), 0x7634706D, FPS, RESOLUTION
+        )
+        depth_out = cv2.VideoWriter(
+            str(DATA_STORE_DIR / "depth_out.mp4"), 0x7634706D, FPS, RESOLUTION
+        )
+
+    data = []
+
+    try:
+        while True:
+            frames = pipeline.wait_for_frames()
+            depth_frame = frames.get_depth_frame()
+            color_frame = frames.get_color_frame()
+
+            if not depth_frame or not color_frame:
+                continue
+
+            depth_image = np.asanyarray(depth_frame.get_data())
+            color_image = np.asanyarray(color_frame.get_data())
+
+            depth_colormap = cv2.applyColorMap(
+                cv2.convertScaleAbs(depth_image, alpha=0.03), cv2.COLORMAP_JET
+            )
+
+            depth_colormap_dim = depth_colormap.shape
+            color_colormap_dim = color_image.shape
+
+            if depth_colormap_dim != color_colormap_dim:
+                color_image = cv2.resize(
+                    color_image,
+                    dsize=(depth_colormap_dim[1], depth_colormap_dim[0]),
+                    interpolation=cv2.INTER_AREA,
+                )
```

Comment:
I think this resize is not necessary because we want to record color image in its original resolution

Reply:
Right, I've done it because initialy I stacked both depth and color into one image. I forgot to remove this part while separating the display into two windows. I'll correct this